File: N1978.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.lang.Math;
import java.util.StringTokenizer;

public class N1978 {
	public static void main(String[] args) throws NumberFormatException, IOException {
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		
		int N = Integer.parseInt(br.readLine());
		int result = 0;

		StringTokenizer st = new StringTokenizer(br.readLine());
		for(int i=0; i<N; i++)
		{
			int Num = Integer.parseInt(st.nextToken());
			if(prime(Num))
			{ result++; }
		}
		
		System.out.println(result);
	}
	
	static boolean prime (int num)
	{
		boolean check = true;
		if(num==1) { return false; }
		
		for (int i=2; i<=Math.sqrt(num); i++)
		{
			if(num%i==0)
			{ check = false; break; }
			else {}
		}
		
		return check;
	}
}

File: N1978.kt
import java.io.*;
import java.util.*;

private val bw = BufferedWriter(OutputStreamWriter(System.`out`))
var n: Int = 0
var ans: Int = 0

fun main(args: Array<String>) {
    init()
    solution()
}

private fun init() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    n = br.readLine().toInt()

    val st = StringTokenizer(br.readLine())
    for (x in 1..n step 1) {
        var num: Int = st.nextToken().toInt()
        if (isPrime(num))
            ans++
    }
    bw.write(ans.toString())
}

private fun solution() {

    bw.flush()
    bw.close()
}

fun isPrime(num: Int): Boolean {
    if (num == 1) return false
    for (x in 2..Math.sqrt(num.toDouble()).toInt() step 1) {
        if (num % x == 0) return false
    }
    return true
}
